app/src/main/java/com/marverenic/music/activity/instance/ArtistActivity.java

```diff
@@ -155,18 +155,20 @@ private void setLastFmReference(LfmArtist lfmArtist) {
         mLfmReference = lfmArtist;
         mRelatedArtists = new ArrayList<>();
 
-        for (LfmArtist relatedArtist : lfmArtist.getSimilarArtists()) {
-            mMusicStore.findArtistByName(relatedArtist.getName())
-                    .subscribe(
-                            found -> {
-                                if (found != null) {
-                                    mRelatedArtists.add(relatedArtist);
-                                    setupAdapter();
-                                }
-                            },
-                            throwable -> {
-                                Timber.e(throwable, "Failed to find artist");
-                            });
+        if (lfmArtist.getSimilarArtists() != null) {
```

Comment:
This was in response to a non-fatal crash that appeared in Crashlytics.

This doesn't fix the problem since Last.fm sends an empty array of similar artists (rather than `null`), so the null check is unnecessary. The problem was actually caused by Last.fm not having an entry and is returning a `null` as a `LfmArtist` to `ArtistActivity.setLastFmReference(LfmArtist)`. This means that `lfmArtist.getSimilarArtists()` was throwing the NPE – not the foreach loop over a potentially null array.